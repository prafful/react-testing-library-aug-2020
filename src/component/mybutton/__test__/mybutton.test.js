import React from 'react';
import ReactDOM from 'react-dom'
import MyButton  from '../mybutton'
import { render, cleanup } from '@testing-library/react';
import renderer from "react-test-renderer";

import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

it('mybutton renders without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<MyButton></MyButton>, div)
})


it('check if button renders in correct way', ()=>{
    const {getByTestId} = render(<MyButton label='click me'></MyButton>)
    expect(getByTestId('button')).toHaveTextContent('Hello click me')
})


it('check if button renders in correct way', ()=>{
    const {getByTestId} = render(<MyButton label='add'></MyButton>)
    expect(getByTestId('button')).toHaveTextContent('add')
})

it('create snapshot if it does not exist and match it for all test in future', ()=>{
    const tree = renderer.create(<MyButton label='snapshot'></MyButton>).toJSON()
    expect(tree).toMatchSnapshot()
})