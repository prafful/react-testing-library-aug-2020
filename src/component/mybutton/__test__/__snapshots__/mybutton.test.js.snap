// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot if it does not exist and match it for all test in future 1`] = `
<div
  className="button"
  data-testid="button"
>
  Hello 
  snapshot
   
</div>
`;
